### PR TITLE
Improved `MetaInfConfigurationTest` cases

### PR DESCRIPTION
**Pull request by joakime**
Establish baseline tests for MetaInfConfiguration in preparation for PRs that adjust lookup / scan behaviors of `MetaInfConfiguration`

I have a PR that uses Resource better and with Java 8 stream APIs (requested by @gregw) pending.
Just wanted to establish what the `MetaInfConfiguration` behaviors are now, so that the new PR is easier to review.

### CODE REVIEW

File: jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/MetaInfConfiguration.java [hunk 1]
```diff
@@ -293,7 +293,10 @@ protected void scanJars(WebAppContext context) throws Exception
         //only look for fragments if web.xml is not metadata complete, or it version 3.0 or greater
         List<String> scanTypes = new ArrayList<>(__allScanTypes);
         if (context.getMetaData().isMetaDataComplete() || (context.getServletContext().getEffectiveMajorVersion() < 3) && !context.isConfigurationDiscovered())
+        {
             scanTypes.remove(METAINF_FRAGMENTS);
+            scanTypes.remove(METAINF_RESOURCES);
```
Comment (janbartel) [inline jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/MetaInfConfiguration.java]: The spec definitely says we should not look at `META-INF/web-fragment.xml`, however I don't see anywhere where it says that we shouldn't look for `META-INF/resources` or `META-INF tlds`. Can you point to the text that would support this change?

**Review by janbartel — CHANGES_REQUESTED** [review]

File: jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/MetaInfConfiguration.java (hunk 1 above)
Comment (joakime) [reply jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/MetaInfConfiguration.java]: TLDs are not impacted by the above change, so lets ignore that part of your question.

`META-INF/web-fragment.xml` and `META-INF/resources/` are both introduced in Servlet 3.0 - didn't exist prior, only reason.
In fact, `META-INF/resources/` didn't exist until _after_ the Servlet 3.0 public draft (that's how late it was added) - it only showed up in the Servlet 3.0 final spec.

Comment (janbartel) [reply jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/MetaInfConfiguration.java]: According to the spec `metadata-complete` means that everything is declared in `web.xml` (or in a `web-fragment.xml`) and annotations are not to be processed, except for those that have no equivalent in `web.xml`. See section 8.1 pg 71. Static resources have never been declared in `web.xml`, thus `metadata-complete` should have no impact on them.

Comment (janbartel) [reply jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/MetaInfConfiguration.java]: Potentially, maybe we should be excluding static resources from fragment jars that have been excluded by an ordering in `web.xml/web-fragment.xml`. Currently I think they will be added to the context base resource regardless of if they're excluded or not. We do the addition of `META-INF/resources` in `MetaInfConfiguration` before we have parsed `web.xml` and the `web-fragment.xmls` to find out whether there are any excluded fragments.  Of course, you can always use a jetty `org.eclipse.jetty.server.webapp.WebInfIncludeJarPattern` to exclude those jars from being examined and thus prevent their resources from being added to the context base resource.

Again, unfortunately the spec doesn't say anything about exclusion of static resources (or `tlds` for that matter) from jars that have been excluded by an ordering - it just refers to annotation processing and `web-fragment.xml`.

Comment (joakime) [reply jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/MetaInfConfiguration.java]: Went ahead and rolled this change back.

**Review by janbartel — CHANGES_REQUESTED** [review]
If you're now programmatically creating the various web.xml/web-fragment.xml/tlds etc you should delete those files that are checked in.

**Comment by joakime** [comment]
> If you're now programmatically creating the various web.xml/web-fragment.xml/tlds etc you should delete those files that are checked in.

Consider them removed.

**Review by janbartel — APPROVED** [review]

**Pull request merged**